src/main/java/net/hydromatic/sqllogictest/Main.java

```diff
@@ -40,14 +39,15 @@ public class Main {
   private Main() {}
 
   public static void main(String[] argv) throws IOException {
-    execute(true, System.out, System.err, argv);
+    ExecutionOptions options = new ExecutionOptions(
+            true, System.out, System.err);
+    execute(options, argv);
   }
 
-  /** As {@link #main} but does not call {@link System#exit} if {@code exit}
-   * is false. */
-  public static int execute(boolean exit, PrintStream out, PrintStream err,
-      String... argv) throws IOException {
-    ExecutionOptions options = new ExecutionOptions(exit, out, err);
+  /** Execute the program using the specified command-line options. */
+  public static int execute(
+          ExecutionOptions options,
+          String... argv) throws IOException {
```

Comment:
If we pass `options` then it doesn't make much sense to also pass `argv`; we can do the `argv` parsing before hand.

Reply:
You use the options to register various command-line options, new executors, and only after that you can call "parse". The way the parsing of argv is handled changes based on the options and executors that you have registered.
We could bundle argv within the ExecutionOptions indeed, but the parsing of the options still has to be done using a separate call.

Reply:
`options` and `argv` represent the same thing so it is kinda weird to have both. Anyways this is not a blocking comment so feel free to ignore.

Reply:
I will rename the ExecutionOptions class to ExecutionOptionsParser. This should make its role clearer.